### DISCUSSION TITLE
Flow payload?

### DISCUSSION THREAD

**Discussion by mluethi** [nfstream/nfstream#52] [Q&A]
Hi there!

NFStream is a great library!
I'm looking for a way to get the payload of a TCP flow. I'd like to do an analysis of the first x bytes of this payload.
Thank you very much for any help!

Regard,
Marcel

**Comment by neyney10** [comment]
Hello! 

I'm using the first n bytes of a TCP stream as well as features for my ML, creating a new NFPlugin and then extracting the payload via `packet.ip_packet` doesn't solve the issue? I've used a library such as scapy to extract more features from it somewhat like this:

IP(packet.ip_packet) and then u can access all higher layers such as TCP, DNS and such.
you can get the payload of the tcp like this: 
```
scapy_ip_packet = IP(packet.ip_packet)
scapy_ip_packet[TCP].payload # IS THIS WHAT YOU ARE LOOKING FOR? it returns array of bytes.
```

Or, it could be that u are talking about TCP reassembly? which is totally a different topic.

**Comment by drnpkr** [comment]
Hi @mluethi 

splt_analysis is destined exactly for this. It specifies the sequence of first packets length for early statistical analysis.

Please see the documentation. By default, it is disabled. If you want the first 10 packet lengths, you set splt_analysis=10
